source/includes/inkling-reference/_algorithms.html.md

```diff
@@ -0,0 +1,196 @@
+# Advanced: Algorithms
+
+> Syntax (in the global scope)
+
+```inkling--syntax
+algorithm <localAlgorithmId>
+   is <globalAlgorithmId>
+   [
+     <parameterName> => <literal>
+     [ ‘,’ <parameterName> => <literal> ]*
+   ]?
+end
+```
+
+> Example (in the global scope)
+
+```Inkling--code
+schema GameState
+   Float32 position,
+   Float32 velocity,
+   Float32 angle,
+   Float32 rotation
+end
+
+schema Action
+   Float32 command
+end
+
+schema CartPoleConfig
+   Int8 episode_length,
+   UInt8 deque_size
+end
+
+simulator the_simulator(CartPoleConfig)
+   action (Action)
+   state (GameState)
+end
+
+
+algorithm My_TRPO_Settings
+   is TRPO
+   learning_rate => 5,
+   network_size => 48
+end
+
+concept balance is estimator
+   predicts (Action)
+   follows input(GameState)
+   feeds output
+
+end
+
+curriculum balance_curriculum
+   train balance
+   using algorithm My_TRPO_Settings
+      # optional to override parameter with same type
+      learning_rate => 0.5
+   end
+
+   with simulator the_simulator
+   objective open_ai_gym_default_objective
+
+       lesson balancing
+           configure
+               constrain episode_length with Int8{-1},
+               constrain deque_size with UInt8{1}
+           until
+               maximize open_ai_gym_default_objective
+end
+```
+
+
+> Syntax (in the curriculum)
+
+```inkling-syntax
+using algorithm <algorithmIdentifier>
+   [
+     <parameterName> => <literal>
+     [ ‘,’ <parameterName> => <literal> ]*
+   end
+]?
+```
+
+> Example (in the curriculum)
+
+```inkling-code
+schema GameState
+   Float32 position,
+   Float32 velocity,
+   Float32 angle,
+   Float32 rotation
+end
+
+schema Action
+   Float32 command
+end
+
+schema CartPoleConfig
+   Int8 episode_length,
+   UInt8 deque_size
+end
+
+simulator the_simulator(CartPoleConfig)
+   action (Action)
+   state (GameState)
+end
+
+concept balance is estimator
+   predicts (Action)
+   follows input(GameState)
+   feeds output
+
+end
+
+curriculum balance_curriculum
+   train balance
+   using algorithm TRPO
+      learning_rate => 5
+      network_size => 48
+   end
+
+   with simulator the_simulator
+   objective open_ai_gym_default_objective
+
+       lesson balancing
+           configure
+               constrain episode_length with Int8{-1},
+               constrain deque_size with UInt8{1}
+           until
+               maximize open_ai_gym_default_objective
+end
+```
+
+The algorithm clause is for advanced users who want to experiment changing machine learning algorithms and their training parameters. The clause provides syntax either within the curriculum statement to force a choice of algorithm for that curriculum, or it can be specified outside of a curriculum if used in multiple concepts (and therefore multiple lessons). It is also possible to override a global parameter inside of a curriculum if needed.
+
+### Valid Algorithm Identifiers
+
+* `DQN` (Deep Q Learning) - classification only
```

Comment:
Curious if we need full name for all algorithms. In case need it, I listed their names here :)

DDPG (Deep Deterministic Policy Gradients)
DiscreteDDPG (Deep Deterministic Policy Gradients with Discrete Actions)
Q-Table (Q Learning with Q Table)
TRPO (Trust Region Policy Optimization)

Reply:
It's very helpful, thank you Ruofan! I added them.